src/main/java/com/github/codeidoscope/ServerConnection.java

```diff
@@ -0,0 +1,58 @@
+package com.github.codeidoscope;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class ServerConnection implements ServerConnectionWrapper {
+    private ServerSocket serverSocket;
+    private Socket socket;
+    InputStream inputStream;
+    OutputStream outputStream;
+    private int portNumber;
+
+    @Override
+    public void openConnection(int portNumber) {
+        this.portNumber = portNumber;
+        try {
+            serverSocket = new ServerSocket(portNumber);
+            socket = serverSocket.accept();
+            inputStream = socket.getInputStream();
+            outputStream = socket.getOutputStream();
+        } catch (IOException e) {
+            throw new RuntimeException();
+        }
+    }
+
+    @Override
+    public String getInput() {
+        try {
+            return new BufferedReader(new InputStreamReader(inputStream)).readLine();
+        } catch (IOException e) {
+            throw new RuntimeException();
+        }
```

Comment:
Same here about catching and re-throwing a different exceptions. Exceptions are your friend, no need to hide them 😃.

Reply:
@mylesmegyesi The reasoning was that `IOException`s are checked and `Runtime Exception`s are unchecked, so I could let it bubble up that way (and I was told it was good practice to do so, but if that's not the case I guess I'll go read up some more on them!)

Reply:
There are pros and cons. Obviously, exceptions propagating everywhere would not be pretty. But, we can cross that bridge once it gets ugly.

I usually solve the problem by creating an error boundary where those checked exceptions are caught and handled or logged appropriately.